# Issue Investigation

**This notebook is used to look into potential issues that arose when importing data. We encountered an issue where the data reported NA values in the `citation_issued` column in a few instances, and others in some other cases.**

### Set-up and Imports

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## St Paul and Nashville

> These were the first two issue we encountered

#### Load Data and Investigate

In [3]:
 # Set the path to st paul
st_paul_path = "../00_source_data/mn_saintpaul_2020_04_01.csv"

# set the path to nashville
nashville_path = "../01_remove_data/tn_nashville_2020_04_01.csv"

In [17]:
# Read in both csvs
stp_df = pd.read_csv(st_paul_path, usecols=[
                "date",
                "time",
                "type",
                "citation_issued",
            ])
nash_df = pd.read_csv(nashville_path, usecols=[
                "date",
                "time",
                "type",
                "citation_issued",
            ])

/var/folders/79/hh7cth4s3p3dbtdpc8j1m9bm0000gn/T/ipykernel_97156/90807190.py:2: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  stp_df = pd.read_csv(st_paul_path, usecols=[
/var/folders/79/hh7cth4s3p3dbtdpc8j1m9bm0000gn/T/ipykernel_97156/90807190.py:8: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  nash_df = pd.read_csv(nashville_path, usecols=[


In [18]:
# Missing citations in stp_df
stp_df.citation_issued.isna().sum()

3

In [19]:
# missing citations in nash_df
nash_df.citation_issued.isna().sum()

320

In [20]:
# total missing in stp
stp_df.isna().sum()

date               0
time               0
type               0
citation_issued    3
dtype: int64

In [21]:
# see the 3 missing values in stp
stp_df[stp_df.citation_issued.isna()]

date      time       type citation_issued
31756   2014-01-17  20:51:00  vehicular             NaN
126268  2014-03-03  01:17:00  vehicular             NaN
649854  2013-12-14  09:06:00  vehicular             NaN

> The only rows we care about that are NA are the 3x citation_issued outcomes. We can drop these rows and continue with our analysis. These three stops were vehicular, but appear to have no other data from the investigation.

In [22]:
# total missing in nash
nash_df.isna().sum()

date                  0
time               5467
type                  0
citation_issued     320
dtype: int64

> The 320 missing values are a bigger problem in the Nashville data. We will drop this city from the analysis.

## Winston-Salem

In [25]:
win_path = "../00_source_data/nc_winston-salem_2020_04_01.csv"
win_df = pd.read_csv(win_path, usecols=[
                "date",
                "time",
                "type",
                "citation_issued",
            ])

In [26]:
win_df.head()

date      time       type  citation_issued
0  2000-01-11  23:30:00  vehicular            False
1  2000-01-12  00:59:00  vehicular             True
2  2000-01-12  21:05:00  vehicular            False
3  2000-01-12  21:05:00  vehicular             True
4  2000-01-12  22:37:00  vehicular            False

In [27]:
# show mising values
win_df.isna().sum()

date                   0
time               96776
type                   0
citation_issued        0
dtype: int64

## Mesa AZ

In [37]:
mesa_path = "../00_source_data/az_mesa_2023_01_26.csv"
mesa_df = pd.read_csv(win_path,)

/var/folders/79/hh7cth4s3p3dbtdpc8j1m9bm0000gn/T/ipykernel_97156/2831451292.py:2: DtypeWarning: Columns (23) have mixed types. Specify dtype option on import or set low_memory=False.
  mesa_df = pd.read_csv(win_path,)


In [39]:
mesa_df.columns

Index(['raw_row_number', 'date', 'time', 'location', 'county_name',
       'subject_age', 'subject_race', 'subject_sex', 'officer_id_hash',
       'department_name', 'type', 'arrest_made', 'citation_issued',
       'warning_issued', 'outcome', 'contraband_found', 'contraband_drugs',
       'contraband_weapons', 'frisk_performed', 'search_conducted',
       'search_person', 'search_vehicle', 'search_basis', 'reason_for_frisk',
       'reason_for_search', 'reason_for_stop', 'raw_Ethnicity', 'raw_Race',
       'raw_action_description'],
      dtype='object')

In [33]:
mesa_df.isna().sum()

date               0
type               0
citation_issued    0
dtype: int64

> Mesa is giving big problems on the import, its gone